File: cpp/771. 宝石与石头.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        unordered_set<char> jewelsSet;
        int ans=0;
        int n=jewels.length(),m=stones.length();
        for(int i=0;i<n;i++){
            jewelsSet.insert(jewels[i]);
        }
        for(int i=0;i<m;i++){
            if(jewelsSet.count(stones[i])){
                ans++;
            }
        }
        return ans;
    }
};

File: 刷题/golang/771. 宝石与石头.go
func numJewelsInStones(jewels, stones string) (ans int) {
    // 把 jewels 转换成字符集合 mask
    mask := 0
    for _, c := range jewels {
        mask |= 1 << (c & 63)
    }
    // 统计有多少 stones[i] 在集合 mask 中
    for _, c := range stones {
        ans += mask >> (c & 63) & 1
    }
    return
}